### ISSUE TITLE
Link flag inconsistent with WCAG 2.1 AA

### ISSUE THREAD

**Issue by FederalCircuitCompendium** [itmaybejj/editoria11y#51]
The following flag is inconsistent with WCAG 2.1 AA: 
          
          This link's text is:
          [hyperlink text]
          
          [Links should be meaningful and concise](https://webaim.org/techniques/hypertext/link_text). Readers often skim by link titles. This is especially true of screen reader users, who navigate using a list of on-page links.
          
          A linked URL breaks this pattern; the reader has to read the preceding paragraph to figure out the link's purpose from context.
          
              Meaningful and concise link: "Tips for writing meaningful links"
              Linked URL, as pronounced by a screen reader: "H T T P S colon forward-slash forward-slash example dot com forward-slash tips forward-slash meaningful-links"

WCAG 2.1 AA does not require that links be embedded into "meaningful and concise" language. To the contrary, it requires that the purpose of every link be determinable from the link text alone or from the link text plus the surrounding context. 

In many instances, the actual link text is necessary, particularly where the content may be printed out. For example, the link www.github.com cannot meaningfully be replaced with just Github if it is going to be printed, as the critical information be lost. This is especially true if there is a path associated with a website. 

Rather, this flag should follow WCAG 2.1, not invent a requirement that is inconsistent with it.

**Comment by itmaybejj** [comment]
A couple notes here before we decide if changes are needed:

1. That test only throws if the link starts with "https://," exactly because of the exception you list. The assumption in writing the test is that someone who left "https://" at the start of a link probably does not know what they are doing, and a tooltip is in order. "github.com" would not be flagged.
2. I recommend configuring this check as a dismissable "Manual Check," not error. The intention is for you to review and then dismiss it ("Mark OK") if it meets the exceptions you list. If you are seeing it as an alert that cannot be marked OK, please check your site configuration. I can help if you tell me whether you are using Drupal, WordPress or a custom implementation.
3. This tool encourages best practices for legibility, which includes many tests that go beyond what WCAG requires. In the library and in Drupal, tests can be turned off. This functionality will probably come to WordPress later this year.

This is what that tip currently looks like in v3 dev:

<img width="648" height="335" alt="alert window, text follows" src="https://github.com/user-attachments/assets/76eb5428-f763-4d1a-9189-c84fda43ca42" />

> Link text should not be a URL

> To fix Change this link to the title of its destination or purpose.

> Readers skim for links by name. This is especially true of screen reader users, who navigate and search using a list of on-page links. Linked URLs cannot be skimmed or found by in-page searches, so people are less likely to find them and less likely to click them.
